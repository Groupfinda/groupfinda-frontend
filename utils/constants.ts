export const eventCategories = [
  "Outdoors",
  "Adventure",
  "Technology",
  "Health",
  "Sports",
  "Fitness",
  "Learning",
  "Food & Drink",
  "Culture",
  "Music",
  "Social",
  "Games",
  "Arts",
  "Pets",
  "Hobbies",
  "Volunteer",
];
