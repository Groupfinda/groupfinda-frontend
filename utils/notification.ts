import Constants from 'expo-constants'
import * as Permissions from 'expo-permissions'
import { Platform } from "react-native";
import * as Notifications from 'expo-notifications'

export async function registerForPushNotificationsAsync() {
    let token;
    if (Constants.isDevice) {
        const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
            finalStatus = status;
        }

        if (finalStatus !== 'granted') {
            alert("Notifications allow us to send you updates on group matches and chat messages!")
            return
        }

        token = (await Notifications.getExpoPushTokenAsync()).data
        console.log(token)
    }

    if (Platform.OS === 'android') {
        Notifications.setNotificationChannelAsync('default', {
            name: 'default',
            importance: Notifications.AndroidImportance.MAX,
            vibrationPattern: [0, 250, 250, 250],
            lightColor: "#FF231F7C"
        })
    }

    return token
}