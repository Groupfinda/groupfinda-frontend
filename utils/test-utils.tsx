import React from "react";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { myTheme } from "../custom-theme";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { MockedProvider, MockedResponse } from "@apollo/react-testing";
import { render, RenderAPI, RenderOptions } from "react-native-testing-library";

const AllProviders = (apolloMocks: MockedResponse[]) => ({
  children,
}: {
  children: JSX.Element;
}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...myTheme }}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          {children}
        </MockedProvider>
      </ApplicationProvider>
    </>
  );
};

type CustomRenderType = (
  component: React.ReactElement<any>,
  options?: RenderOptions,
  apolloProps?: MockedResponse[]
) => RenderAPI;

const customRender: CustomRenderType = (
  ui,
  options,
  apolloProps: MockedResponse[] = []
) => render(ui, { wrapper: AllProviders(apolloProps), ...options });

export * from "react-native-testing-library";
export { customRender as render };
