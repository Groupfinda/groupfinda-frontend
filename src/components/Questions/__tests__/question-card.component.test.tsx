import React from "react";
import { render, cleanup, fireEvent, act } from "../../../../utils/test-utils";
import { QuestionCard } from "../extra/question-card.component";

afterEach(cleanup);

let props = {
    order: 1,
    question: "Dummy question",
    selectedValue: 0,
    setSelectedValue: jest.fn()
}

test("renders UI correctly", () => {
    const tree = render(<QuestionCard {...props}/>);
    expect(tree.toJSON()).toMatchSnapshot();
})

test("answer change handled", () => {
    const component = render(<QuestionCard {...props}/>);
    
    const accurateOption = component.getByText("Accurate");
    expect(accurateOption).toBeTruthy();
    expect(accurateOption).toHaveProp("status", "basic");
    fireEvent.press(accurateOption);
    expect(accurateOption).toHaveProp("status", "control");
    fireEvent.press(accurateOption);
    expect(accurateOption).toHaveProp("status", "basic");

    const slightlyAccurateOption = component.getByText("Slightly Accurate");
    expect(slightlyAccurateOption).toBeTruthy();
    expect(slightlyAccurateOption).toHaveProp("status", "basic");
    fireEvent.press(slightlyAccurateOption);
    expect(slightlyAccurateOption).toHaveProp("status", "control");
    fireEvent.press(slightlyAccurateOption);
    expect(slightlyAccurateOption).toHaveProp("status", "basic");

    const neutralOption = component.getByText("Neutral");
    expect(neutralOption).toBeTruthy();
    expect(neutralOption).toHaveProp("status", "basic");
    fireEvent.press(neutralOption);
    expect(neutralOption).toHaveProp("status", "control");
    fireEvent.press(neutralOption);
    expect(neutralOption).toHaveProp("status", "basic");

    const slightlyInaccurateOption = component.getByText("Slightly Inaccurate");
    expect(slightlyAccurateOption).toBeTruthy();
    expect(slightlyInaccurateOption).toHaveProp("status", "basic");
    fireEvent.press(slightlyInaccurateOption);
    expect(slightlyInaccurateOption).toHaveProp("status", "control");
    fireEvent.press(slightlyInaccurateOption);
    expect(slightlyInaccurateOption).toHaveProp("status", "basic");

    const inaccurateOption = component.getByText("Inaccurate");
    expect(inaccurateOption).toBeTruthy();
    expect(inaccurateOption).toHaveProp("status", "basic");
    fireEvent.press(inaccurateOption);
    expect(inaccurateOption).toHaveProp("status", "control");
    fireEvent.press(inaccurateOption);
    expect(inaccurateOption).toHaveProp("status", "basic");
})