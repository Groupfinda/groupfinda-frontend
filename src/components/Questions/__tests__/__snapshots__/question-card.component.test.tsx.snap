// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#006ECB",
        "borderColor": "#E4E9F2",
        "borderRadius": 4,
        "borderWidth": 1,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "height": 0,
        }
      }
    />
    <View
      style={
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <View
        appearance="default"
        level="1"
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
            },
            undefined,
          ]
        }
      >
        <Text
          appearance="default"
          category="h6"
          status="control"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "fontSize": 18,
                "fontWeight": "800",
              },
              Object {
                "backgroundColor": "#006ECB",
              },
            ]
          }
        >
          1
          . 
          Dummy question
        </Text>
      </View>
    </View>
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          undefined,
        ]
      }
    />
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Object {
            "backgroundColor": "#CBF5FE",
          },
        ]
      }
    >
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="white"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#0052AA",
              },
            ]
          }
        >
          Accurate
        </Text>
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="white"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#0052AA",
              },
            ]
          }
        >
          Slightly Accurate
        </Text>
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="white"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#0052AA",
              },
            ]
          }
        >
          Neutral
        </Text>
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="white"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#0052AA",
              },
            ]
          }
        >
          Slightly Inaccurate
        </Text>
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={20}
            bbWidth={20}
            fill="white"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#0052AA",
              },
            ]
          }
        >
          Inaccurate
        </Text>
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
